lib/stdio/type.js

```diff
@@ -3,14 +3,10 @@ import {isUint8Array} from './utils.js';
 
 // The `stdin`/`stdout`/`stderr` option can be of many types. This detects it.
 export const getStdioOptionType = (stdioOption, optionName) => {
-	if (isAsyncGenerator(stdioOption)) {
+	if (isGenerator(stdioOption)) {
 		return 'generator';
 	}
 
-	if (isSyncGenerator(stdioOption)) {
-		throw new TypeError(`The \`${optionName}\` option must use an asynchronous generator, not a synchronous one.`);
-	}
```

Comment:
The previous syntax required the generators to be `async` due to implementation details. This new syntax does not.